util/synoptic.py

```diff
@@ -14,20 +14,20 @@ class SynopticUtils(object):
 
     SCHEMA = "{http://www.isis.stfc.ac.uk//instrument}"
 
-    def __init__(self, config_repo_path):
+    def __init__(self, config_repo_path: str) -> None:
         self.synoptics_path = os.path.join(config_repo_path, "configurations", "synoptics")
 
-    def _prefix_schema(self, tag):
+    def _prefix_schema(self, tag: str) -> str:
         return "{schema}{tag}".format(schema=SynopticUtils.SCHEMA, tag=tag)
 
-    def get_synoptics_filenames(self):
+    def get_synoptics_filenames(self) -> list[str]:
         return [
             f
             for f in CommonUtils.get_directory_contents_as_list(self.synoptics_path)
             if f.endswith(".xml")
         ]
 
-    def get_type_target_pairs(self, synoptic_xml):
+    def get_type_target_pairs(self, synoptic_xml: str) -> list[str]:
         """
         Returns a set of type, target pairs used in this synoptic
```

Comment:
Just purely from the docs this hint doesn't look quite right, the function says it returns `type, target pairs` so I'd expect something like `list[tuple[str, str]]` as the type hint?

Reply:
Thanks Tom  - should have spotted that!  Especially from line 50:
`result.append((type.text, target.text))`